examples/sir_hmc.py

```diff
@@ -478,20 +478,24 @@ def predict(args, data, samples, truth=None):
     particle_plate = pyro.plate("particles", args.num_samples, dim=-1)
 
     # First we sample discrete auxiliary variables from the continuous
-    # variables sampled in vectorized_model. Here infer_discrete runs a
-    # forward-filter backward-sample algorithm. We'll add these new samples to
-    # the existing dict of samples.
+    # variables sampled in vectorized_model. This samples only time steps
+    # [0:duration]. Here infer_discrete runs a forward-filter backward-sample
+    # algorithm. We'll add these new samples to the existing dict of samples.
     model = poutine.condition(continuous_model, samples)
     model = particle_plate(model)
+    if args.dct:  # Apply the same reparameterizer as during inference.
+        rep = DiscreteCosineReparam()
+        model = poutine.reparam(model, {"S_aux": rep, "I_aux": rep})
```

Comment:
This was the bug: `S_aux` was being sampled from the prior because the inferred samples only provided `S_aux_dct`.